docs/contracts.rst

```diff
@@ -488,6 +488,27 @@ Functions can be declared ``view`` in which case they promise not to modify the
 .. warning::
   The compiler does not enforce yet that a ``view`` method is not modifying state.
 
+.. _pure-functions:
+
+**************
+Pure Functions
+**************
+
+Functions can be declared ``pure`` in which case they promise not to read from or modify the state.
```

Comment:
Perhaps we should explain in more detail what reading from or writing to the state means.